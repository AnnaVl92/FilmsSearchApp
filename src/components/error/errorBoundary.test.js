import React from 'react';
import { shallow, mount } from "enzyme";
import expect from 'expect';
import ErrorBoundary from './errorBoundary.jsx';

describe('<ErrorBoundary />', () => {
    const ComponentWithError = () => null;
    const wrapper = shallow(
        <ErrorBoundary>
            <ComponentWithError />
        </ErrorBoundary>
    );

    it('displays error message on error generated by child', () => {
        wrapper.setState({
            error: 'error name',
            errorInfo: 'error info'
        });
        expect(wrapper.state().error).toEqual('error name');
        expect(wrapper.state().errorInfo).toEqual('error info');
        expect(wrapper).toMatchSnapshot();
        console.log(wrapper.debug() );    
    });
});
