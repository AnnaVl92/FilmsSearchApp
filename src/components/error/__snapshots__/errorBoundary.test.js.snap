// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> displays error message on error generated by child 1`] = `
<Fragment>
  <div>
    error name
  </div>
  <div />
</Fragment>
`;
